coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1114,6 +1115,50 @@ func (s *server) notifyWorkspaceBuildFailed(ctx context.Context, workspace datab
 	}
 }
 
+func (s *server) notifyWorkspaceManualBuildFailed(ctx context.Context, workspace database.Workspace, build database.WorkspaceBuild) {
+	templateAdmins, err := s.Database.GetUsers(ctx, database.GetUsersParams{
```

Comment:
Do all template admins own all templates?

Reply:
@johnstcn I may need your RBAC expert skills.

What I could do is to call `GetUsers( RbacRole: TemplateAdmin )` , then call `GetOrganizationIDsByMemberIDs( IDs from previous call )`, and remove template admins that do not belong to `workspace.OrganizationID`.

Question, is there an easy alternative solution?

Reply:
@marcin Do we also need to take into account template-level ACLs?

Reply:
For now, I would limit it to generic template admins as with ACL you may mark anyone as a template admin, and accidentally notify everyone?

<img width="1379" alt="Screenshot 2024-08-27 at 12 59 35" src="https://github.com/user-attachments/assets/1f5193e4-075a-453d-98ce-8fb0f0a4b6d1">

Reply:
OK -- I was going to suggest something like

```pseudo-go
GetUsers(RbacRole: TemplateAdmin) | { | user | api.Authorizer.Authorize(ctx, user, policy.ActionUpdate, template }`
```

However, this would take into account the template-level user and group ACLs. So the approach you're suggesting above would make more sense in this case.

Reply:
Ah, so this could be even simpler! Maybe we should consider ACLs if it is a quick win

Reply:
@johnstcn there might be a soft dependency blocker. I need to introduce `Authorizer` to `provisionerdserver`. It should be technically possible, but I don't know if it does not make ProvisionerdServer more dependent on API (cycle).

Reply:
Fixed